main.tf

```
@@ -1,26 +1,33 @@
 resource "azurerm_resource_group" "cache-resourcegroup" {
   name     = "${var.product}-cache-${var.env}"
-  location = "${var.location}"
+  location = var.location
 
-  tags = "${merge(var.common_tags,
+  tags = merge(var.common_tags,
     map("lastUpdated", "${timestamp()}")
-    )}"
+  )
 }
 
-data "template_file" "redistemplate" {
-  template = "${file("${path.module}/templates/redis-paas.json")}"
-}
 
-resource "azurerm_template_deployment" "redis-paas" {
-  template_body       = "${data.template_file.redistemplate.rendered}"
+resource "azurerm_redis_cache" "redis" {
   name                = "${var.product}-${var.env}"
-  resource_group_name = "${azurerm_resource_group.cache-resourcegroup.name}"
-  deployment_mode     = "Incremental"
+  location            = azurerm_resource_group.cache-resourcegroup.location
+  resource_group_name = azurerm_resource_group.cache-resourcegroup.name
+  capacity            = var.capacity
+  family              = "P"
+  sku_name            = "Premium"
+  subnet_id           = var.subnetid
+  enable_non_ssl_port = false
+  minimum_tls_version = var.minimum_tls_version
+
+  redis_configuration {
+    maxmemory_reserved              = var.maxmemory_reserved
+    maxfragmentationmemory_reserved = var.maxfragmentationmemory_reserved
+    maxmemory_delta                 = var.maxmemory_delta
+    maxmemory_policy = var.maxmemory_policy
+  }
 
-  parameters = {
-    cachename = "${var.product}-${var.env}"
-    location  = "${azurerm_resource_group.cache-resourcegroup.location}"
-    subnetid  = "${var.subnetid}"
-    env       = "${var.env}"
+  tags = {
```

this should just pass through common tags,
ARM template resources don't do it as it's too much effort updating them all the time

```suggestion
  tags = var.common_tags
```